core/frontend/src/CheckpointConnection.ts

```diff
@@ -54,13 +54,55 @@ export class CheckpointConnection extends IModelConnection {
     const accessToken = await IModelApp.getAccessToken();
     const changeset = await IModelApp.hubAccess.getChangesetFromVersion({ accessToken, iModelId, version });
 
-    let connection: CheckpointConnection;
     const iModelProps = { iTwinId, iModelId, changeset };
+    if (IpcApp.isValid) // when called from Ipc, use the Ipc implementation
+      return this.openFromIpc(iModelProps);
+
+    const routingContext = IModelRoutingContext.current || IModelRoutingContext.default;
+    const connection = new this(await this.callOpen(iModelProps, routingContext), false);
+    RpcManager.setIModel(connection);
+    connection.routingContext = routingContext;
+    RpcRequest.notFoundHandlers.addListener(connection._reopenConnectionHandler);
+
+    IModelConnection.onOpen.raiseEvent(connection);
+    return connection;
+  }
+
+  /**
+   * Open a readonly IModelConnection to a Checkpoint of an iModel from Ipc.
+   * @note this function is equivalent to [[openRemote]] but allows more flexibility specifying the changeset version,
+   * and also optionally starting a prefetch operation.
+   */
+  public static async openFromIpc(args: OpenCheckpointArgs): Promise<CheckpointConnection> {
+    const connection = new this(await IpcApp.appFunctionIpc.openCheckpoint(args), true);
+    IModelConnection.onOpen.raiseEvent(connection);
+    return connection;
+  }
+
+  /**
```

Comment:
there must have been a bad merge.

Reply:
Sorry I was trying to git commit amend the email associated with two previous commits. Accidentally made this bad merge, now it's resolved.